consensus/core/src/core.rs

```diff
@@ -1145,6 +1181,105 @@ mod test {
         }
     }
 
+    #[tokio::test]
+    async fn test_leader_schedule_change() {
+        telemetry_subscribers::init_for_testing();
+        let default_params = Parameters::default();
+
+        // create the cores and their signals for all the authorities
+        let mut cores = create_cores(vec![1, 1, 1, 1]);
+
+        // Now iterate over a few rounds and ensure the corresponding signals are created while network advances
+        let mut last_round_blocks = Vec::new();
+        for round in 1..=30 {
+            let mut this_round_blocks = Vec::new();
+
+            for (core, signal_receivers, block_receiver, _, _) in &mut cores {
+                // Wait for min round delay to allow blocks to be proposed.
+                sleep(default_params.min_round_delay).await;
```

Comment:
If possible, this test should use
```
#[tokio::test(flavor = "current_thread", start_paused = true)]
```
to allow fast forwarding time during sleep. Also, does `sleep()` once before the loop work?

Reply:
Leaving this open for now, need to investigate why using those test decorators causes the test to fail but move d the sleep outside of the loop

```
thread 'core::test::test_leader_schedule_change' panicked at consensus/core/src/core.rs:1219:22:
called `Result::unwrap()` on an `Err` value: Elapsed(())
stack backtrace:
   0: rust_begin_unwind
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/result.rs:1649:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/result.rs:1073:23
   4: consensus_core::core::test::test_leader_schedule_change::{{closure}}
             at ./src/core.rs:1217:29
   5: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/future/future.rs:125:9
   6: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/future/future.rs:125:9
   7: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:659:57
   8: tokio::runtime::coop::with_budget
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/coop.rs:107:5
   9: tokio::runtime::coop::budget
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/coop.rs:73:5
  10: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:659:25
  11: tokio::runtime::scheduler::current_thread::Context::enter
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:404:19
  12: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:658:36
  13: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:737:68
  14: tokio::runtime::context::scoped::Scoped<T>::set
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context/scoped.rs:40:9
  15: tokio::runtime::context::set_scheduler::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context.rs:176:26
  16: std::thread::local::LocalKey<T>::try_with
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/std/src/thread/local.rs:270:16
  17: std::thread::local::LocalKey<T>::with
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/std/src/thread/local.rs:246:9
  18: tokio::runtime::context::set_scheduler
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context.rs:176:9
  19: tokio::runtime::scheduler::current_thread::CoreGuard::enter
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:737:27
  20: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:646:19
  21: tokio::runtime::scheduler::current_thread::CurrentThread::block_on::{{closure}}
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:175:28
  22: tokio::runtime::context::runtime::enter_runtime
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context/runtime.rs:65:16
  23: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/current_thread/mod.rs:167:9
  24: tokio::runtime::runtime::Runtime::block_on
             at /Users/arunkoshy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/runtime.rs:348:47
  25: consensus_core::core::test::test_leader_schedule_change
             at ./src/core.rs:1197:9
  26: consensus_core::core::test::test_leader_schedule_change::{{closure}}
             at ./src/core.rs:1188:43
  27: core::ops::function::FnOnce::call_once
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/ops/function.rs:250:5
  28: core::ops::function::FnOnce::call_once
             at /rustc/6cf088810f66fff15d05bf7135c5f5888b7c93b4/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
test core::test::test_leader_schedule_change ... FAILED
```

Reply:
The tokio timeout may need to be increased. But just leaving a TODO is ok.